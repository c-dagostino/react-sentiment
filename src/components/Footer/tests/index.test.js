import React from 'react';
import moment from 'moment';
import { shallow } from 'enzyme';

import { Footer } from '../index';

function getWrapper(props) {
  const requiredProps = {
    classes: {
      footer: '/test-footer'
    },
    currentYear: moment('2017-09-15 09:30:00').format('YYYY')
  };

  return shallow(<Footer {...requiredProps} {...props} />);
}

describe('Footer', () => {
  it('Expect component to render properly', () => {
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });
});
