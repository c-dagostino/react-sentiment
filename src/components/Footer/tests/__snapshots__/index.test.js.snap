// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Expect component to render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    classes={
        Object {
            "footer": "/test-footer",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Typography)
        align="center"
        gutterBottom={true}
        variant="title"
>
        © Navis Software 2018
</WithStyles(Typography)>,
      "className": "/test-footer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "center",
        "children": "© Navis Software 2018",
        "gutterBottom": true,
        "variant": "title",
      },
      "ref": null,
      "rendered": "© Navis Software 2018",
      "type": [Function],
    },
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Typography)
          align="center"
          gutterBottom={true}
          variant="title"
>
          © Navis Software 2018
</WithStyles(Typography)>,
        "className": "/test-footer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "center",
          "children": "© Navis Software 2018",
          "gutterBottom": true,
          "variant": "title",
        },
        "ref": null,
        "rendered": "© Navis Software 2018",
        "type": [Function],
      },
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
